### DISCUSSION TITLE
Installed Perfectly on Ubuntu but I cannot install at all on Kali.

### DISCUSSION THREAD

**Discussion by infusion007** [yuezk/GlobalProtect-openconnect#228] [General]
Hi Yuezk,
this Installed Perfectly on Ubuntu 20.10  but I cannot install it at all on Kali 2023.01

Would you have any idea how I could get this to work?

Thanks very much in advance.

**Comment by yuezk** [comment]
I can install it from the source code on Kali 2023.01 in VM. Please try to build from the source code.

### Prepare

```bash
sudo apt-get update
sudo apt-get install -y \
    build-essential \
    qtbase5-dev \
    libqt5websockets5-dev \
    qtwebengine5-dev \
    qttools5-dev \
    qt5keychain-dev \
    openconnect
```

### Build

```bash
cmake -B build \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_CXX_FLAGS_RELEASE=-s

MAKEFLAGS=-j$(nproc) cmake --build build
```

### Install

```bash
sudo cmake --install build

sudo systemctl enable gpservice.service
sudo systemctl daemon-reload
sudo systemctl restart gpservice.service
```

### Run

```
gpclient
```

**Reply by Cyb3rC3lt** [reply]
Thank you, I tested this today and it installed and worked perfectly. Many other clients failed for me. Great job.
